repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java

```diff
@@ -149,7 +151,13 @@ public void deleteEntities(Collection<AtlasVertex> instanceVertices) throws Atla
             deleteTypeVertex(deletionCandidateVertex, isInternalType(deletionCandidateVertex));
 
             if (DEFERRED_ACTION_ENABLED) {
-                createClassificationOnlyPropagationDeleteTasksAndQueue(getAllClassificationEdges(deletionCandidateVertex), RequestContext.get().getDeletedEdgesIds());
+                Set<String> deletedEdgeIds = RequestContext.get().getDeletedEdgesIds();
+                Set<AtlasVertex> classificationVertices = new HashSet<>();
+                for (String deletedEdgeId : deletedEdgeIds) {
+                    AtlasEdge edge = graph.getEdge(deletedEdgeId);
+                    classificationVertices.addAll(GraphHelper.getPropagatableClassifications(edge));
+                }
+                createClassificationOnlyPropagationDeleteTasksAndQueue(classificationVertices, deletedEdgeIds);
```

Comment:
There is a problem here, here we are creating task for each edge and all classification which is very wrong. We need to create task for each edge and classification attached with it. What I mean by that:
In this implementation:
We have edge 1 with classification A,B attached, edge 2 with classification C,D attached. If we follow this approach we will create task for (1,A),(1,B),(1,C),(1,D)(2,A),(2,B),(2,C),(2,D)
But we want to create task for : (1,A),(1,B),(2,C),(2,D)

Reply:
you are correct. can you please check once more ?